pta-intelligent-search-harvester/src/main/java/fi/maanmittauslaitos/pta/search/index/LocalArchiveDocumentSink.java

```diff
@@ -54,11 +54,14 @@ private void writeDocumentToFile(Document doc) throws IOException {
     public int stopIndexing() {
         Path zipFile = Paths.get(outputFileName);
         try {
-            File outputRootDir = zipFile.getParent().toFile();
-            if (!outputRootDir.exists()) {
-                logger.warn("Creating new directory " + outputRootDir);
-                outputRootDir.mkdirs();
+            if (zipFile.getParent() != null) {
```

Comment:
Under what conditions would getParent() return null and if it would, would this object behave as expected?

Reply:
When running the LocalResourceMetadataGenerator without any command line arguments, it uses relative sink file name located in root. Obviously getParent() returns null then.